### PR TITLE
[zen.yandex.ru] Fix download

### PR DESCRIPTION
**Pull request by lksj**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

</details>

Zen.yandex.ru is deprecated, redirect to dzen.ru.
This PR shold fix this issue.

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

### CODE REVIEW

File: yt_dlp/extractor/yandexvideo.py [hunk 1]
```diff
@@ -188,54 +188,43 @@ def _real_extract(self, url):
...
-            'thumbnail': 're:^https://avatars.mds.yandex.net/',
+            'thumbnail': 're:^https://avatars.dzeninfra.ru/',
             'uploader': 'AcademeG DailyStream'
         },
         'params': {
             'skip_download': 'm3u8',
             'format': 'bestvideo',
         },
     }, {
-        'url': 'https://zen.yandex.ru/video/watch/6002240ff8b1af50bb2da5e3',
```
Comment (pukkandan) [inline yt_dlp/extractor/yandexvideo.py:220]: This URL still works in browser, and so should continue to work in yt-dlp

File: yt_dlp/extractor/yandexvideo.py [hunk 2]
```diff
@@ -188,54 +188,43 @@ def _real_extract(self, url):
...
     }, {
-        'url': 'https://zen.yandex.ru/media/id/606fd806cc13cb3c58c05cf5/novyi-samsung-fold-3-moskvich-barahlit-612f93b7f8d48e7e945792a2?from=channel&rid=2286618386.482.1630817595976.42360',
+        'url': 'https://dzen.ru/media/id/606fd806cc13cb3c58c05cf5/novyi-samsung-fold-3-moskvich-barahlit-612f93b7f8d48e7e945792a2?from=channel&rid=2286618386.482.1630817595976.42360',
         'only_matching': True,
     }]
 
     def _real_extract(self, url):
         id = self._match_id(url)
         webpage = self._download_webpage(url, id)
+        webpage = self._download_webpage(url, id)
```
Comment (pukkandan) [inline yt_dlp/extractor/yandexvideo.py:227]: ?

File: yt_dlp/extractor/yandexvideo.py [hunk 3]
```diff
@@ -266,22 +255,23 @@ def _real_extract(self, url):
...
+    _VALID_URL = r'https?://dzen\.ru/(?!media|video)(?:id/)?(?P<id>[a-z0-9-_]+)'
     _TESTS = [{
-        'url': 'https://zen.yandex.ru/tok_media',
+        'url': 'https://dzen.ru/tok_media',
         'info_dict': {
             'id': 'tok_media',
         },
         'playlist_mincount': 169,
     }, {
-        'url': 'https://zen.yandex.ru/id/606fd806cc13cb3c58c05cf5',
```
Comment (pukkandan) [inline yt_dlp/extractor/yandexvideo.py:277]: same

File: yt_dlp/extractor/yandexvideo.py [hunk 4]
```diff
@@ -266,22 +255,23 @@ def _real_extract(self, url):
...
+        'url': 'https://dzen.ru/id/606fd806cc13cb3c58c05cf5',
         'info_dict': {
             'id': '606fd806cc13cb3c58c05cf5',
         },
         'playlist_mincount': 657,
     }]
 
     def _entries(self, id, url):
+        webpage = self._download_webpage(url, id)
         webpage = self._download_webpage(url, id)
```
Comment (pukkandan) [inline yt_dlp/extractor/yandexvideo.py:275]: same

**Review by pukkandan — CHANGES_REQUESTED** [review]

**Comment by lksj** [comment]
Let me explain.
Now zen.yandex.ru redirect to sso.passport.yandex.ru and next redirect to dzen.ru.
Direct dzen.ru redirect to sso.passport.yandex.ru and redirect to dzen.ru again.
Therefore, the download page is not possible. yt-dlp stopped on sso.passport.yandex.ru.
Double redirect is probably temporary.
Why redirect to sso.passport.yandex.ru? Cookies.
I was not able to get and put cookies in a simple way.
Download page and download page again turned out to be the easiest solution.
Yes, my fix is dirty hack, but it work.
I'm ready to refactor the solution, but I need a hint.
What in this case best way for cookies and redirects?

**Comment by pukkandan** [comment]
tests dont pass

**Pull request merged**